src/Illuminate/Collections/Collection.php

```diff
@@ -945,17 +945,27 @@ public function select($keys)
      *
      * @param  int  $count
      * @return static<int, TValue>|TValue|null
+     *
+     * @throws \InvalidArgumentException
      */
     public function pop($count = 1)
     {
-        if ($count === 1) {
-            return array_pop($this->items);
+        if ($count < 0) {
+            throw new InvalidArgumentException('Number of popped items may not be less than zero.');
+        }
+
+        if ($count === 0) {
+            return new static($this->items);
         }
 
         if ($this->isEmpty()) {
             return new static;
         }
```

Comment:
```suggestion
        if ($count === 0 || $this->isEmpty()) {
            return new static;
        }
```

Reply:
`pop()`  is  supposed to return the removed items so when `$count` equals 0, `pop()`  removes no item so it should return an empty collection.